fastly/kv_store.go

```diff
@@ -388,6 +395,67 @@ func (c *Client) GetKVStoreKey(i *GetKVStoreKeyInput) (string, error) {
 	return string(output), nil
 }
 
+// GetKVStoreItemInput is the input to the GetKVStoreItem function.
+type GetKVStoreItemInput struct {
+	// StoreID is the StoreID of the kv store (required).
+	StoreID string
+	// Key is the key of the item to fetch (required).
+	Key string
+}
+
+// GetKVStoreItemOutput is the output of the GetKVStoreItem function.
+type GetKVStoreItemOutput struct {
+	// Value is the value stored in the item.
+	Value io.ReadCloser
+	// Metadata is the metadata stored in the item, if any.
+	Metadata string
+	// Generation is the generation marker of the item.
+	Generation uint64
+}
+
+func (o *GetKVStoreItemOutput) ValueAsBytes() ([]byte, error) {
+	defer o.Value.Close()
+
+	return io.ReadAll(o.Value)
+}
+
+func (o *GetKVStoreItemOutput) ValueAsString() (string, error) {
+	if result, err := o.ValueAsBytes(); err != nil {
+		return "", err
+	} else {
+		return string(result), nil
+	}
+}
```

Comment:
Doc comments would be useful here, mainly to inform the user that the `Value` field is consumed and closed by calling these.